terraso_backend/apps/project_management/graphql/projects.py

```diff
@@ -375,29 +380,22 @@ def mutate_and_get_payload(cls, root, info, project_id, user_id):
         project = cls.get_or_throw(Project, "project_id", project_id)
         user = cls.get_or_throw(User, "user_id", user_id)
         requester = info.context.user
-        requester_membership = project.get_membership(requester)
-        if not requester_membership:
-            cls.not_allowed(
-                MutationTypes.DELETE,
-                msg="User cannot delete other users" " from project where they are not a member",
-            )
+        if user == requester:
+            # leaving a project is implemented as removing yourself
+            if not check_project_permission(requester, "leave", Context(project)):
```

Comment:
question: bit of a general python n00b question: is `Context(project)` the same as `Context(project=project)`? i see both styles used and wanted to confirm they're equivalent

Reply:
Incidentally this works because project is the first field defined in the dataclass, but it's an oversight on my part to not be explicit since this could unexpectedly break in the future.